src/tests/ftest/osa/offline_reintegration.py

```diff
@@ -152,6 +153,9 @@ def run_offline_reintegration_test(self, num_pool, data=False,
                     output = self.dmg_command.pool_reintegrate(self.pool.uuid,
                                                                rank[val],
                                                                "0,2")
+                    # Insert a time delay when trying to reintegrate only few targets
+                    # This will avoid intermittent -2031 when we write data to the object.
+                    time.sleep(5)
```

Comment:
We should not be adding sleeps to our test code.  It would be more useful to retry the operation if we encounter the intermittent -2031 as this would:
1) allow the test to complete quicker if the intermittent issue was not seen
2) provide information as to how often the intermittent issue is seen and if it gets better/worse

Also if we are increasing the run time of the test - either with a sleep or retry - especially in nested for loops, why wasn't the test timeout adjusted?

Reply:
I am not sure whether we encounter only -2031 or I/O error (-2001),etc... So, we cannot check the failure code. We have been different errors. The delay is introduced based on WangDi's input... 

Anyway, I can try to increase the self.print_and_assert_on_rebuild_failure(output) (default is 3 seconds) to self.print_and_assert_on_rebuild_failure(output, timeout=15)... Maybe. that is a better solution...